src/validator.js

```diff
@@ -272,6 +189,70 @@ export default class Validator {
     return output;
   }
 
+  getAddonMetaData() {
+    return this.xpi.getMetaData()
```

Comment:
Yeah - I know what you mean - I'd like to separate concerns as much as possible too. One problem is the collector is needed in the manifest parsers. So to move this would require passing the collector to the xpi class and then from there to the manifest parsers. So there's a bit of a balancing act there.

Also if we move the metadata parsing into the xpi class there's some potential confusion around the xpi meta + the meta from the manifest files. Currently the xpi class has remained pretty dumb in that it only knows and cares about xpi handling which feels right to me.

I think I'll land this now and work with it a bit. But, if we see further scope for refactoring (I feel like there's still a bit of that we can do) then we should do so. One nice thing is it does feel possible to refactor fairly easily since the tests should catch breakage.

Something we could think about is removing the error/notice collection from the metadata parsing and move it up into the validator? I like the idea of the classes only needing to do one job - but then again passing the collector was easy to do and keeps the error collection close to the code it cares about. Another thing with that though is we could de-dupe some of the errors. However there might be a corner to paint ourselves into there as the fields differ quite a bit.